packages/server/src/api/controllers/application.ts

```diff
@@ -626,6 +641,61 @@ export async function importToApp(ctx: UserCtx) {
   ctx.body = { message: "app updated" }
 }
 
+/**
+ * Create a copy of the latest dev application.
+ * Performs an export of the app, then imports from the export dir path
+ */
+export async function duplicateApp(ctx: UserCtx) {
+  const { name: appName, url: possibleUrl } = ctx.request.body
+  const { appId: sourceAppId } = ctx.params
+  const [app] = await dbCore.getAppsByIDs([sourceAppId])
+
+  if (!app) {
+    ctx.throw(404, "Source app not found")
+  }
+
+  const apps = (await dbCore.getAllApps({ dev: true })) as App[]
+
+  checkAppName(ctx, apps, appName)
+  const url = sdk.applications.getAppUrl({ name: appName, url: possibleUrl })
+  checkAppUrl(ctx, apps, url)
+
+  const tmpPath = await sdk.backups.exportApp(sourceAppId, {
+    excludeRows: false,
+    tar: false,
+  })
+
+  // Build a create request that triggers an import from the export path above.
+  const createReq: any = {
+    request: {
+      body: {
+        name: appName,
+        url: possibleUrl,
+        useTemplate: "true",
+        file: {
+          path: tmpPath,
+        },
+      },
+      // Mark the create as a duplicate to kick off the correct event.
+      duplicate: true,
+    },
+  }
+
+  await create(createReq)
```

Comment:
`create` could have its signature changed to `create(ctx: UserCtx<CreateAppRequest, App>)` and then that could flow into here to get rid of the `any` above. In general we're trying not to add any new `any`s into the backend code unless they're unavoidable.